File: backend/app/schemas/analytics.py
from pydantic import BaseModel, ConfigDict


class OverviewMetrics(BaseModel):
    totalSuppliers: int
    avgEsgScore: float
    avgDelayDays: float
    avgDefectRatePct: float
    highRiskCount: int

    model_config = ConfigDict(from_attributes=True)


class CountryDistributionItem(BaseModel):
    country: str
    supplierCount: int

    model_config = ConfigDict(from_attributes=True)


class HistogramBin(BaseModel):
    label: str
    start: float
    end: float
    count: int

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/analytics.ts
import { apiRequest } from "./client";

export interface OverviewMetrics {
  totalSuppliers: number;
  avgEsgScore: number;
  avgDelayDays: number;
  avgDefectRatePct: number;
  highRiskCount: number;
}

export interface CountryDistributionItem {
  country: string;
  supplierCount: number;
}

export interface HistogramBin {
  label: string;
  start: number;
  end: number;
  count: number;
}

export async function getOverviewMetrics(): Promise<OverviewMetrics> {
  return apiRequest<OverviewMetrics>("/analytics/overview");
}

export async function getCountryDistribution(): Promise<CountryDistributionItem[]> {
  return apiRequest<CountryDistributionItem[]>("/analytics/country-distribution");
}

export async function getEsgDistribution(): Promise<HistogramBin[]> {
  return apiRequest<HistogramBin[]>("/analytics/esg-distribution");
}
